base/compiler/optimize.jl

```diff
@@ -58,10 +58,12 @@ struct InliningState{S <: Union{EdgeTracker, Nothing}, MICache, I<:AbstractInter
     interp::I
 end
 
+source_inferred(@nospecialize(src)) = ccall(:jl_ir_flag_inferred, Bool, (Any,), src)
```

Comment:
`is_source_inferred`? Also, the `src` is currently required to be `CodeInfo` or `Vector{UInt8}`. Being Any unspecialized value is not legal in `jl_ir_flag_inferred`.